server_2.R

```diff
@@ -0,0 +1,736 @@
+                     dbname   = "d5m2p6kka0vf8d",
+                     port     = "5432",
+                     host     = "ec2-184-73-232-93.compute-1.amazonaws.com",
+                     sslmode = 'require')
+
+# DB calls - fetch data
+{
+  
+  games  = dbGetQuery(mta_con,
+                      "SELECT *
+                       FROM mta_games
+                       WHERE league = 'League'")
+  
+  events = dbGetQuery(mta_con,
+                      "SELECT e.event_id,
+                              e.date,
+                              e.game_id,
+                              e.player_name,
+                              e.minute,
+                              g.round,
+                              g.league,
+                              g.season,
+                              g.game_result,
+                              g.location,
+                              g.opponent,
+                              g.coach,
+                              p.minutes_played,
+                              p.game_status
+                       FROM mta_events e
+                       INNER JOIN mta_games g ON (g.game_id = e.game_id)
+                       INNER JOIN mta_player_con p ON (p.game_id = e.game_id AND p.player_name = e.player_name)
+                       WHERE g.league = 'League'
+                       ORDER BY e.date, e.minute")
+  
+  players  = dbGetQuery(mta_con,
+                       "SELECT p.*,
+                               g.round,
+                               g.league,
+                               g.season,
+                               g.game_result,
+                               g.location,
+                               g.opponent,
+                               g.coach,
+                               e.event_id
+                        FROM mta_player_con p
+                        INNER JOIN mta_games g ON (g.game_id = p.game_id)
+                        LEFT JOIN mta_events e ON (g.game_id = e.game_id AND p.player_name = e.player_name)
+                        WHERE g.league = 'League'")
+
+  # fetch current league round
+  current_season <- dbGetQuery(mta_con,"SELECT season,
+                                               max(round) as value
+                                        FROM mta_games
+                                        WHERE season IN (
+                                              SELECT max(season) as season
+                                              FROM mta_games
+                                              WHERE round is not null
+                                              AND league = 'League'
+                                          )
+                                        GROUP BY 1;")
+  # fetch coach nick names
+  people <- dbGetQuery(mta_con,"SELECT * 
+                                FROM people")
+}
+
+
+function(input, output, session) {
+
+  # reactive datasets
```

Comment:
at the  # reactive datasets section - this filter repeats itself few times, in general, when you repeat the same operation more than twice in one script it's pretty much means that it deserve a function -   
```
      filter(season %in% input$season_id,
             round <= input$round_id,
             coach %in% input$coach_id,
             opponent %in% input$opponent_id,
             location %in% input$location_id)
```

same notion regarding this one - 
```
      filter(season == current_season$season) %>%
      mutate(points = case_when(game_result == 'W' ~ 3,game_result == 'L' ~ 0,TRUE ~ 1)) %>%
      summarise(goal_scored = sum(mta_score),
                goal_conced = sum(opponent_score),
                league_points = sum(points),
                pct = sum(points)/(n()*3),
                pct_label = paste0(round(pct*100,1),'%'),
                max_round = max(round))
```